benchmark/cmake/modules/AddSwiftBenchmarkSuite.cmake

```diff
@@ -267,6 +267,33 @@ function(add_swift_multisource_nonwmo_benchmark_library objfiles_out)
   set(${objfiles_out} ${objfiles} PARENT_SCOPE)
 endfunction()
 
+function(swift_benchmark_suite_version)
+    # Swift Benchmark Suite version is composed of Swift version and
+    # the git branch version.
+    message("Determining Swift Benchmark Suite version")
+    runcmd(COMMAND "${SWIFT_EXEC}" "--version"
+      VARIABLE SWIFT_VERSION
+      ERROR "Unable to determine Swift version")
+    message("Swift version: ${SWIFT_VERSION}")
+    message("CMake current source dir: ${CMAKE_CURRENT_SOURCE_DIR}")
+    runcmd(COMMAND "git" "-C" "${CMAKE_CURRENT_SOURCE_DIR}" "describe"
+      VARIABLE BRANCH_VERSION
+      ERROR "Unable to determine git tree version")
+    # get_current_git_branch
+    runcmd(COMMAND "git" "-C" "${CMAKE_CURRENT_SOURCE_DIR}" "rev-parse" "--abbrev-ref" "HEAD"
+      VARIABLE BRANCH_VERSION
+      ERROR "Unable to determine git tree version")
+    # get_git_head_ID
+    runcmd(COMMAND "git" "-C" "${CMAKE_CURRENT_SOURCE_DIR}" "rev-parse" "--short" "HEAD"
+      VARIABLE BRANCH_VERSION
+      ERROR "Unable to determine git tree version")
+    message("Git branch ${BRANCH_VERSION}")
```

Comment:
My bad for overwriting the `BRANCH_VERSION` variable. This was just WIP. Only the `runcmd` that does `git describe` goes into that one. 

I want to add 2 more line with branch name and revision hash (because `describe` will pick a nice tag is available), so that we don't need to do [this git foo](https://github.com/apple/swift/blob/41a8b9c084542aaaa8577e26eb89f4de663f50d2/benchmark/scripts/Benchmark_Driver#L145-L157) in Benchmark_Driver, but parse it from `--version` output instead.

Reply:
That isn't what I am talking about. What I am trying to say is that a branch name in a git repository could just be something local. It really doesn't make sense to automatically infer that. Especially if this is something that we want to use to store benchmark suites over time. In such a case, we would want the hash since that is a global source of truth, vs a potential local branch name.

Reply:
The branch name is used as directory to [`log_results`](https://github.com/apple/swift/blob/41a8b9c084542aaaa8577e26eb89f4de663f50d2/benchmark/scripts/Benchmark_Driver#L165) into and these directories are later used in [`Benchmark_Driver compare`](https://github.com/apple/swift/blob/41a8b9c084542aaaa8577e26eb89f4de663f50d2/benchmark/scripts/Benchmark_Driver#L308). The individual log results are stored in these branch-named directories and include the [git revision in their names](https://github.com/apple/swift/blob/41a8b9c084542aaaa8577e26eb89f4de663f50d2/benchmark/scripts/Benchmark_Driver#L182).

I though it would make sense to remove Bechmark_Driver's dependency on `git` and get all the needed info from `Bechmark_O --version`. That's why I wanted to put all currently required git provided string in there. Do you think this isn't a goal worth pursuing?